pyroomacoustics/room.py

```diff
@@ -1530,26 +1548,69 @@ def plot_rir(self, select=None, FD=False):
-            else:  # one column
-                plt.subplot(len(pairs), 1, k + 1)
-            if not FD:
-                plt.plot(np.arange(len(h)) / float(self.fs), h)
-            else:
-                u.real_spectrum(h)
-            plt.title("RIR: mic" + str(r) + " source" + str(s))
-            if r == n_mic - 1:
+        if select is None:
+            fig, axes = plt.subplots(n_mic, n_src)
+            for r in range(n_mic):
+                for s in range(n_src):
+                    h = self.rir[r][s]
+                    if not FD:
+                        axes[r, s].plot(np.arange(len(h)) / float(self.fs / 1000), h)
+                    else:
+                        axes[r, s].specgram(h, Fs=self.fs / 1000)
+
+                    if r == 0:
+                        axes[r, s].set_title("Source {}".format(s), fontsize="medium")
+                    if s == n_src - 1:
+                        axes[r, s].annotate(
+                            "Mic {}".format(r),
+                            xy=(1.02, 0.5),
+                            xycoords="axes fraction",
+                            rotation=270,
+                            ha="left",
+                            va="center",
+                        )
+                    elif s == 0 and FD:
+                        axes[r, s].set_ylabel("Freq. [kHz]")
+
+                    if r < n_mic - 1:
+                        axes[r, s].set_xticks([])
+                    else:
+                        axes[r, s].set_xlabel("Time [ms]")
+                    if s > 0:
+                        axes[r, s].set_yticks([])
+            fig.align_ylabels(axes[:, 0])
+            fig.tight_layout()
+
+        else:
+            fig, axes = plt.subplots(len(pairs), 1, squeeze=False)
+            for k, (r, s) in enumerate(pairs):
+                h = self.rir[r][s]
                 if not FD:
-                    plt.xlabel("Time [s]")
+                    axes[k,0].plot(np.arange(len(h)) / float(self.fs / 1000), h)
                 else:
-                    plt.xlabel("Normalized frequency")
+                    axes[k,0].specgram(h, Fs=self.fs / 1000)
 
-        plt.tight_layout()
+                if len(pairs) == 1:
+                    axes[k, 0].set_title("Mic {}, Source {}".format(r, s))
+                else:
+                    axes[k,0].annotate(
+                        "M{}, S{}".format(r, s),
+                        xy=(1.02, 0.5),
+                        xycoords="axes fraction",
+                        rotation=270,
+                        ha="left",
+                        va="center",
+                    )
+
+                if FD:
+                    axes[k,0].set_ylabel("Freq. [kHz]")
+
+                if k < len(pairs) - 1:
+                    axes[k,0].set_xticks([])
+                else:
+                    axes[k,0].set_xlabel("Time [ms]")
+            fig.align_ylabels(axes[:, 0])
+            fig.tight_layout()
```

Comment:
Should we return `fig` and `axes` in case the user would like to make additional changes?